packages/node_modules/@webex/react-container-activity-list/src/selectors.js

```diff
@@ -6,12 +6,37 @@ import {
   ITEM_TYPE_DAY_SEPARATOR,
   ITEM_TYPE_NEW_MESSAGE_SEPARATOR
 } from '@webex/react-component-activity-list';
+import {OrderedMap} from 'immutable';
 
 import defaultFormatters from './formatters';
 import {isActivityVisible} from './helpers';
 
+const getConversationThreadActivities = (state) => state.conversation.get('threadActivities');
+const getConversationNonThreadActivities = (state) => state.conversation.get('sortNonThreadActivities');
+const getConversationActivities = createSelector(
+  [
+    getConversationThreadActivities,
+    getConversationNonThreadActivities
+  ],
+  (
+    threadActivities = {},
+    sortNonThreadActivities = []
+  ) => {
+    const finalList = [];
+    let start = 0;
+
+    for (let i = 0; i < sortNonThreadActivities.length; i += 1) {
```

Comment:
This is some awesome magic! Can you throw a quick comment in about what this is doing?

Reply:
I had an old comment similar to this:

> Can you add a comment stating assumptions of these lists? e.g. are they supposed to be sorted (and by what vector)? How is the threaded activities connected to non-thread activities, etc..
This is assuming for someone that may have not spent as much time as you did understanding the convo service.